azurerm/data_source_batch_account.go

```diff
@@ -0,0 +1,74 @@
+package azurerm
+
+import (
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/services/batch/mgmt/2017-09-01/batch"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func dataSourceArmBatchAccount() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceArmBatchAccountRead,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateAzureRMBatchAccountName,
+			},
+			"resource_group_name": resourceGroupNameDiffSuppressSchema(),
+			"location":            locationForDataSourceSchema(),
+			"storage_account_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"pool_allocation_mode": {
+				Type:             schema.TypeString,
+				Optional:         true,
+				Default:          string(batch.BatchService),
+				DiffSuppressFunc: ignoreCaseDiffSuppressFunc,
+				ValidateFunc: validation.StringInSlice([]string{
+					string(batch.BatchService),
+					string(batch.UserSubscription),
+				}, true),
```

Comment:
Hmm, I am not sure to understand why this field should be marked as Computed? It should be specified in the configuration.

Reply:
Ok sorry, I was looking on the resource, not the datasource :-)

Reply:
so this is specific to the Data Source where whilst this information will be returned it can't be set (since we only obtain the Batch Account using the Name & Resource Group - the other fields are outputs)